lib/internal/http.js

```diff
@@ -31,9 +32,30 @@ function ondrain() {
   if (this._httpMessage) this._httpMessage.emit('drain');
 }
 
+function now() {
+  const hr = process.hrtime();
+  return hr[0] * 1000 + hr[1] / 1e6;
+}
+
+class HttpRequestTiming extends PerformanceEntry {
+  constructor(statistics) {
+    super();
+    this.name = 'HttpRequest';
+    this.entryType = 'http';
+    this.duration = statistics.endTime - statistics.startTime;
```

Comment:
Maybe instead of doing this we can just set `startTime`, remove `endTime`, and just do something like this here:

```js
const diff = process.hrtime(statistics.startTime);
this.duration = diff[0] * 1000 + diff[1] / 1e6;
```

This simplifies things a bit and avoids any loss in precision.

Reply:
Indeed, i didn't know that we could compute the diff by passing an old value to the `hrtime`. I've updated it and removed the call to `now()`